File: client/src/app/modules/quotation/services/mail.redirection.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { MailRedirectionType } from '../../quotation/model/MailRedirectionType';

@Injectable({
  providedIn: 'root'
})
export class MailRedirectionTypeService extends AppRestService<MailRedirectionType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getMailRedirectionTypes() {
    return this.getListCached(new HttpParams(), "mail-redirection-types");
  }

  addOrUpdateMailRedirectionType(mailRedirectionType: MailRedirectionType) {
    this.clearListCache(new HttpParams(), "mail-redirection-types");
    return this.addOrUpdate(new HttpParams(), "mail-redirection-type", mailRedirectionType, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/MailRedirectionTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.MailRedirectionType;

public interface MailRedirectionTypeService {
    public List<MailRedirectionType> getMailRedirectionTypes();

    public MailRedirectionType getMailRedirectionType(Integer id);

    public MailRedirectionType addOrUpdateMailRedirectionType(MailRedirectionType mailRedirectionType);
}
